SpeechSDK 1.22.0

mzachar opened the pull request:
Hi,

I update the binding for SDK version 1.21.0.

- Improved multi-platform support in build.rs (so it will build on arm / aarch64 and link correct libraries)
- Added rust bindings for EmbeddedSpeechConfig (similar to c++ headers). However it is not yet publicly available but the SDK contains the headers for it already

adambezecny commented:
hi,

thanks for the PR. I will definitely have a look but bear in mind it will take some time (quite busy these days...:( ). At a first glance it is quite big, maybe we should split it into multiple separate PRs, but let me review first. I'll keep you posted. Thanks for the effort!

mzachar commented:
EmbeddedSpeechConfig allows you to run speech recognition or synthesis offline locally on the computer without the need for internet access to the azure cloud.

It looks like the official go binding is quite behind as official Java/.NET/CPP already contains classes to this "bleeding edge" functionality (however to used them you need to have access to private preview from Microsoft)

Java:
https://docs.microsoft.com/en-us/java/api/com.microsoft.cognitiveservices.speech.embeddedspeechconfig?view=azure-java-stable

.NET:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.cognitiveservices.speech.embeddedspeechconfig?view=azure-dotnet 

CPP:
https://docs.microsoft.com/en-us/cpp/cognitive-services/speech/embeddedspeechconfig

So if you have access you can use this rust bindings to play with it. The only difference in the code is that you provide EmbeddedSpeechConfig instead of SpeechConfig - the rest of the API works the same


If you want to split the pull request in some parts just let me know.
Basically I did 3 things:
- updated build.rs to support multi-platform builds
   still linux only but it will now compile correctly with different processor architectures
   
- Updated the rust bindings based on the changes in the SDK 1.21.0
   There was some C methods renamed, some got extra arguments.
   Error handling as the errors now return pointers to error structure instead of error number (should be similar as in go).
   Most of the changes in the pull request are based on the new generated bindings.rs from SDK 1.21.0
   
- Added [embedded_speech_config.rs](https://github.com/jabber-tools/cognitive-services-speech-sdk-rs/pull/3#diff-7913768266b6b3b884a147d8e39fb9f50f1b86ff959cb12201900435670ca09e) to support new "bleeding edge" api with [embedded_recognize_once_async_from_file.rs](https://github.com/jabber-tools/cognitive-services-speech-sdk-rs/pull/3#diff-146f4329f237aaf684e02b3813f2438e998d7c489c93a899dab12b61424ade2a) example.
   Note: you need to have access to the private preview to run that example
   

Just take your time I'm not in hurry :)

adambezecny commented:
hi,

let's split this as outlined above:

PR1 - updated build only to support additional architectures
PR2 - updated bindings to support SDK 1.22 (ideally)
PR3 - embedded speech config related changes

It will be much easier to review and test for me.

regards,

Adam

mzachar commented:
Once #4 will be merged I will rebase main onto it and we can continue with just with the SDK 1.22.0 upgrade I will then create the last pull request with embedded speech support

mzachar commented:
It took a while but I have the changes ready. Please take a look

adambezecny commented:
hi Matej,

welcome back from holiday :) Give me couple of days to review, I will try to be more responsive this time but still need some time.

adambezecny commented on src/speech/speech_synthesis_word_boundary_event.rs:
```diff
@@ -35,6 +41,7 @@ impl SpeechSynthesisWordBoundaryEvent {
                     synthesizer_event_handle_release,
                 ),
                 audio_offset,
+                duration_ms,
```
api **synthesizer_word_boundary_event_get_values** now returns also **boundary_type**.  Why this new parameter was not added to struct **SpeechSynthesisWordBoundaryEvent** same as **duration_ms**?

mzachar replied:
Good catch looks like I completely forgot about it

adambezecny commented:
I think this really looks great now. Before I merge just a quick one: did you rerun all examples? are they working fine? I compiled the examples right now but found out my API key is expired (and so is my free Azure subscription:( ). So just confirm please examples work fine at your end with new API, I will believe you :)

mzachar commented:
I run the examples in "recognizer" and "synthesizer" folders but I was not able to verify the "dialog" examples as I get the following error:
`Invalid Cognitive Servi Please verify the provided subscription details and language information`

I guess my subscription does not include the BotFramework

adambezecny commented:
ok, let's do the last PR now. thanks!
